src/main/java/io/github/fablabsmc/fablabs/api/fluidvolume/v1/math/Fraction.java

```diff
@@ -1,48 +1,51 @@
-package io.github.fablabsmc.fablabs.api.fluidvolume.v1;
+package io.github.fablabsmc.fablabs.api.fluidvolume.v1.math;
 
-import static com.google.common.base.Preconditions.checkArgument;
-import static com.google.common.math.IntMath.gcd;
-
-import java.util.Objects;
 import java.util.stream.IntStream;
 
 import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.math.IntMath;
 import com.mojang.datafixers.Dynamic;
 import com.mojang.datafixers.types.DynamicOps;
 
+import net.minecraft.nbt.CompoundTag;
 import net.minecraft.util.DynamicSerializable;
 
+/**
+ * a number whose value is represented as 2 integers, a numerator and denominator.
```

Comment:
I think I might rewrite this class sometime soon to store whole buckets as a separate number, since that can be spun out during simplification and would mean we aren't at risk of losing precision at higher values. I'll do that in another PR once this is merged.

Reply:
I could just make them longs and then :crab: problem is gone

Reply:
grondag's suggesting whole numbers *and* longs

Reply:
Should be rewritten when we decide. For now it is fine

Reply:
yeah, I just want to put down a note for future PRs.

Reply:
why would you need more than a long numerator and denominator :concern:

Reply:
go ask them yourself

Reply:
probably make them mixed numbers at some point

Reply:
@l-Luna why?